iis-workflows/iis-workflows-ingest-pmc/src/test/java/eu/dnetlib/iis/workflows/ingest/pmc/metadata/PmcXmlHandlerTest.java

```diff
@@ -58,35 +60,91 @@ public void testParsing() throws Exception {
+		try {
+			SAXParserFactory saxFactory = SAXParserFactory.newInstance();
+			saxFactory.setValidating(false);
+			SAXParser saxParser = saxFactory.newSAXParser();
+			XMLReader reader = saxParser.getXMLReader();
+			reader.setFeature("http://xml.org/sax/features/validation", false);
+			reader.setFeature("http://apache.org/xml/features/nonvalidating/load-dtd-grammar", false);
+			reader.setFeature("http://apache.org/xml/features/nonvalidating/load-external-dtd", false);
+
+			ExtractedDocumentMetadata.Builder metaBuilder = ExtractedDocumentMetadata.newBuilder();
+			metaBuilder.setId("some-id");
+			metaBuilder.setText("");
+			PmcXmlHandler pmcXmlHandler = new PmcXmlHandler(metaBuilder);
+			saxParser.parse(inputStream = PmcXmlHandler.class.getResourceAsStream(filePath), pmcXmlHandler);
+			checkJats10Metadata(metaBuilder.build());
 		} finally {
-			if (inputStream!=null) {
+			if (inputStream != null) {
 				inputStream.close();
 			}
 		}
 	}
-	
+
+	private void checkJats10Metadata(ExtractedDocumentMetadata meta) throws Exception {
+		assertEquals("research-article", meta.getEntityType());
+		assertEquals("BMC Systems Biology", meta.getJournal());
+		assertEquals(1, meta.getExternalIdentifiers().size());
+		assertEquals("19943949", meta.getExternalIdentifiers().get("pmid"));
+		assertEquals("111", meta.getPages().getStart());
+		assertEquals("111", meta.getPages().getEnd());
+
+		assertNotNull(meta.getReferences());
+		assertEquals(34, meta.getReferences().size());
+		// checking first reference
+		assertEquals(1, meta.getReferences().get(0).getPosition().intValue());
+		assertEquals(
+				"Koonin, E. Comparative genomics, minimal gene-sets and the last universal common ancestor. Nat Rev Microbiol. 2003; 1 (2): 127-136",
+				meta.getReferences().get(0).getText());
+		assertEquals("Comparative genomics, minimal gene-sets and the last universal common ancestor",
+				meta.getReferences().get(0).getBasicMetadata().getTitle());
+		assertEquals(1, meta.getReferences().get(0).getBasicMetadata().getAuthors().size());
+		assertEquals("Koonin, E", meta.getReferences().get(0).getBasicMetadata().getAuthors().get(0));
+		assertEquals("127", meta.getReferences().get(0).getBasicMetadata().getPages().getStart());
+		assertEquals("136", meta.getReferences().get(0).getBasicMetadata().getPages().getEnd());
+		assertEquals("Nat Rev Microbiol", meta.getReferences().get(0).getBasicMetadata().getSource());
+		assertEquals("1", meta.getReferences().get(0).getBasicMetadata().getVolume());
+		assertEquals("2003", meta.getReferences().get(0).getBasicMetadata().getYear());
+		assertEquals("2", meta.getReferences().get(0).getBasicMetadata().getIssue());
+		assertEquals(2, meta.getReferences().get(0).getBasicMetadata().getExternalIds().size());
+		assertEquals("10.1038/nrmicro751", meta.getReferences().get(0).getBasicMetadata().getExternalIds().get("doi"));
+		assertEquals("15035042", meta.getReferences().get(0).getBasicMetadata().getExternalIds().get("pmid"));
```

Comment:
Extracting `meta.getReferences().get(0).getBasicMetadata()` to variable could help in reading this